addons/auth_signup/res_users.py

```diff
@@ -212,7 +213,8 @@ def signup(self, cr, uid, values, token=None, context=None):
             if partner_user:
                 # user exists, modify it according to values
                 values.pop('login', None)
-                values.pop('name', None)
+                if single_email_re.match(partner_user.name) == None:
```

Comment:
We are inviting users just by `email id` and because of name and log in are required. 
I put same email id in name and provide a way to change the name when user *accept the invitation*
Currently we are not allowing user to change name by popping name from submitted data. So here i allowed name change for user if it's old name was email id.